src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs

```diff
@@ -165,31 +166,19 @@ public object Clone()
             return name;
         }
 
-        private static Func<string, AssemblyName>? s_getAssemblyName;
-        private static Func<string, AssemblyName> InitGetAssemblyName()
-        {
-            Type readerType = Type.GetType(
-                    "System.Reflection.Metadata.MetadataReader, System.Reflection.Metadata",
-                    throwOnError: true)!;
-
-            MethodInfo? getAssemblyNameMethod = readerType.GetMethod(
-                "GetAssemblyName",
-                BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static,
-                null,
-                [typeof(string)],
-                null) ??
-                throw new MissingMethodException(readerType.FullName, "GetAssemblyName");
-            return s_getAssemblyName = getAssemblyNameMethod.CreateDelegate<Func<string, AssemblyName>>();
-        }
-
         /*
          * Get the AssemblyName for a given file. This will only work
          * if the file contains an assembly manifest. This method causes
          * the file to be opened and closed.
          */
         public static AssemblyName GetAssemblyName(string assemblyFile)
         {
-            return (s_getAssemblyName ?? InitGetAssemblyName())(assemblyFile);
+            return GetName(null, assemblyFile);
+
+            [UnsafeAccessor(UnsafeAccessorKind.StaticMethod, Name = "GetAssemblyName")]
+            static extern AssemblyName GetName(
+                [UnsafeAccessorType("System.Reflection.Metadata.MetadataReader, System.Reflection.Metadata")] object? _,
+                string assemblyFile);
```

Comment:
```suggestion
            return GetAssemblyNameInternal(null, assemblyFile);

            [UnsafeAccessor(UnsafeAccessorKind.StaticMethod, Name = "GetAssemblyName")]
            static extern AssemblyName GetAssemblyNameInternal(
                [UnsafeAccessorType("System.Reflection.Metadata.MetadataReader, System.Reflection.Metadata")] object? _,
                string assemblyFile);
```

The pattern we use in SPCL when method names conflict or won't work is to append `Internal`.